### ISSUE TITLE
CardDav not working with Mac OS X Contacts application

### ISSUE THREAD

**Issue by alex3137** [owncloud/core#23910]
### Steps to reproduce
1. Configure in OS X Contacts app a Carddav account

### Expected behaviour
The account is added and contacts are synchronised in the application 

### Actual behaviour
The account is added, contacts are not synchronised and the application returns an error: The operation couldn’t be completed. (CoreDAVHTTPStatusErrorDomain error 405.)

### Server configuration
**Operating system**: Debian 8.4

**Web server:** Apache 2.4.10

**Database:** SQLlite

**PHP version:** PHP 5.6.19

**ownCloud version:** 9.0.1

**Updated from an older ownCloud or fresh install:** Fresh install

**Where did you install ownCloud from:** Official Debian package

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**List of activated apps:**

```
Enabled:
  - activity: 2.2.1
  - calendar: 1.0
  - comments: 0.2
  - contacts: 1.1.0.0
  - dav: 0.1.6
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0
Disabled:
  - encryption
  - external
  - files_external
  - user_external
  - user_ldap
```

**The content of config/config.php:**

```
{
    "system": {
        "updatechecker": false,
        "instanceid": "oc8pou1n9s2p",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "owncloud.besnard.me"
        ],
        "datadirectory": "\/var\/www\/owncloud\/data",
        "overwrite.cli.url": "https:\/\/owncloud.besnard.me",
        "dbtype": "sqlite3",
        "version": "9.0.1.3",
        "logtimezone": "UTC",
        "installed": true
    }
}
```

**Are you using external storage, if yes which one:** no

**Are you using encryption:** Encryption is provided by my HAProxy front-end

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Apple Contacts 9.0

**Operating system:** Mac OS X 10.11.4

#### Web server error log
```
10.10.10.100 - - [11/Apr/2016:11:14:59 +0000] "PROPFIND /remote.php/dav/addressbooks/users/Alex/contacts/ HTTP/1.1" 401 1470 "-" "Mac OS X/10.11.4 (15E65) AddressBook/1679.9"
10.10.10.100 - Alex [11/Apr/2016:11:14:59 +0000] "PROPFIND /remote.php/dav/addressbooks/users/Alex/contacts/ HTTP/1.1" 207 1949 "-" "Mac OS X/10.11.4 (15E65) AddressBook/1679.9"
10.10.10.100 - - [11/Apr/2016:11:14:59 +0000] "OPTIONS /remote.php/dav/principals/users/Alex/ HTTP/1.1" 401 1476 "-" "Mac OS X/10.11.4 (15E65) AddressBook/1679.9"
10.10.10.100 - Alex [11/Apr/2016:11:15:00 +0000] "OPTIONS /remote.php/dav/principals/users/Alex/ HTTP/1.1" 200 1359 "-" "Mac OS X/10.11.4 (15E65) AddressBook/1679.9"
10.10.10.100 - - [11/Apr/2016:11:15:02 +0000] "GET /ocs/v2.php/apps/notifications/api/v1/notifications?format=json HTTP/1.1" 200 808 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.11; rv:45.0) Gecko/20100101 Firefox/45.0"
10.10.10.100 - Alex [11/Apr/2016:11:15:05 +0000] "PROPFIND /remote.php/dav/principals/users/Alex/ HTTP/1.1" 207 2687 "-" "Mac OS X/10.11.4 (15E65) AddressBook/1679.9"
10.10.10.100 - - [11/Apr/2016:11:15:05 +0000] "PROPFIND /remote.php/dav/addressbooks/users/Alex/ HTTP/1.1" 401 1472 "-" "Mac OS X/10.11.4 (15E65) AddressBook/1679.9"
10.10.10.100 - Alex [11/Apr/2016:11:15:05 +0000] "PROPFIND /remote.php/dav/addressbooks/users/Alex/ HTTP/1.1" 207 5239 "-" "Mac OS X/10.11.4 (15E65) AddressBook/1679.9"
10.10.10.100 - - [11/Apr/2016:11:15:06 +0000] "PROPFIND /.well-known/carddav HTTP/1.1" 301 577 "-" "Mac OS X/10.11.4 (15E65) AddressBook/1679.9"
10.10.10.100 - - [11/Apr/2016:11:15:06 +0000] "PROPFIND / HTTP/1.1" 405 1000 "-" "Mac OS X/10.11.4 (15E65) AddressBook/1679.9"
10.10.10.100 - - [11/Apr/2016:11:15:07 +0000] "PROPFIND /principals/ HTTP/1.1" 405 541 "-" "Mac OS X/10.11.4 (15E65) AddressBook/1679.9"
10.10.10.100 - - [11/Apr/2016:11:15:35 +0000] "PROPFIND /.well-known/carddav HTTP/1.1" 301 577 "-" "Mac OS X/10.11.4 (15E65) AddressBook/1679.9"
10.10.10.100 - - [11/Apr/2016:11:15:35 +0000] "PROPFIND / HTTP/1.1" 405 1004 "-" "Mac OS X/10.11.4 (15E65) AddressBook/1679.9"
10.10.10.100 - - [11/Apr/2016:11:15:35 +0000] "PROPFIND /principals/ HTTP/1.1" 405 541 "-" "Mac OS X/10.11.4 (15E65) AddressBook/1679.9"
```

#### ownCloud log (data/owncloud.log)
```
{"reqId":"JCvIGEFm6hozpSGdJ12v","remoteAddr":"10.10.10.100","app":"no app in context","message":"Exception: {\"Exception\":\"DomainException\",\"Message\":\"Contacts tables are missing. Nothing to do.\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/migration\\\/migrateaddressbooks.php(83): OCA\\\\Dav\\\\Migration\\\\AddressBookAdapter->setup()\\n#1 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/application.php(186): OCA\\\\Dav\\\\Migration\\\\MigrateAddressbooks->setup()\\n#2 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/install.php(26): OCA\\\\Dav\\\\AppInfo\\\\Application->migrateAddressbooks()\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(639): include('\\\/var\\\/www\\\/ownclo...')\\n#4 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(590): OC_Installer::includeAppScript('\\\/var\\\/www\\\/ownclo...')\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(561): OC_Installer::installShippedApp('dav')\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/setup.php(370): OC_Installer::installShippedApps()\\n#7 \\\/var\\\/www\\\/owncloud\\\/core\\\/controller\\\/setupcontroller.php(64): OC\\\\Setup->install(Array)\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(832): OC\\\\Core\\\\Controller\\\\SetupController->run(Array)\\n#9 \\\/var\\\/www\\\/owncloud\\\/index.php(39): OC::handleRequest()\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/migration\\\/addressbookadapter.php\",\"Line\":72}","level":3,"time":"2016-04-11T10:19:50+00:00","method":"POST","url":"\/index.php","user":"--"}
{"reqId":"JCvIGEFm6hozpSGdJ12v","remoteAddr":"10.10.10.100","app":"no app in context","message":"Exception: {\"Exception\":\"DomainException\",\"Message\":\"Calendar tables are missing. Nothing to do.\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/migration\\\/migratecalendars.php(84): OCA\\\\Dav\\\\Migration\\\\CalendarAdapter->setup()\\n#1 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/application.php(202): OCA\\\\Dav\\\\Migration\\\\MigrateCalendars->setup()\\n#2 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/install.php(27): OCA\\\\Dav\\\\AppInfo\\\\Application->migrateCalendars()\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(639): include('\\\/var\\\/www\\\/ownclo...')\\n#4 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(590): OC_Installer::includeAppScript('\\\/var\\\/www\\\/ownclo...')\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(561): OC_Installer::installShippedApp('dav')\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/setup.php(370): OC_Installer::installShippedApps()\\n#7 \\\/var\\\/www\\\/owncloud\\\/core\\\/controller\\\/setupcontroller.php(64): OC\\\\Setup->install(Array)\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(832): OC\\\\Core\\\\Controller\\\\SetupController->run(Array)\\n#9 \\\/var\\\/www\\\/owncloud\\\/index.php(39): OC::handleRequest()\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/migration\\\/calendaradapter.php\",\"Line\":68}","level":3,"time":"2016-04-11T10:19:50+00:00","method":"POST","url":"\/index.php","user":"--"}
```

**Comment by DeepDiver1975** [comment]
same reason as explained in https://github.com/owncloud/core/issues/23908#issuecomment-208290775

**Comment by pucicu** [comment]
I don't see how this problem is solved by the #23908. I also have not found anywhere a solution how to fix this problem (except some lengthy explanations with some vague or not-working suggestions). What are the steps to solve this issue here? Thanks!
